tests/unit_tests/test_ladder_service.py

```diff
@@ -187,53 +187,96 @@ async def test_start_game_called_on_match(
     ladder_service.start_game.assert_called_once()
 
 
-async def test_choose_map(ladder_service: LadderService):
-    ladder_service.get_ladder_history = CoroutineMock(
-        return_value=[1, 2, 3]
+async def test_start_game_map_selection(
+    ladder_service: LadderService, player_factory
+):
+    p1 = player_factory(
+        ladder_rating=(1500, 500),
+        ladder_games=0,
+    )
+    p2 = player_factory(
+        ladder_rating=(1000, 100),
+        ladder_games=1000,
     )
 
-    ladder_service.game_service.ladder_maps = [
-        (1, "some_map", "maps/some_map.v001.zip"),
-        (2, "some_map", "maps/some_map.v001.zip"),
-        (3, "some_map", "maps/some_map.v001.zip"),
-        (4, "CHOOSE_ME", "maps/choose_me.v001.zip"),
-    ]
+    queue = ladder_service.queues["ladder1v1"]
+    queue.map_pools.clear()
+    newbie_map_pool = mock.Mock()
+    full_map_pool = mock.Mock()
+    queue.add_map_pool(newbie_map_pool, None, 500)
+    queue.add_map_pool(full_map_pool, 500, None)
 
-    chosen_map = await ladder_service.choose_map([None])
+    await ladder_service.start_game(p1, p2)
 
-    # Make the probability very low that the test passes because we got lucky
-    for _ in range(20):
-        assert chosen_map == (4, "CHOOSE_ME", "maps/choose_me.v001.zip")
+    newbie_map_pool.choose_map.assert_called_once()
+    full_map_pool.choose_map.assert_not_called()
 
 
-async def test_choose_map_all_maps_played(ladder_service: LadderService):
-    ladder_service.get_ladder_history = CoroutineMock(
-        return_value=[1, 2, 3]
+async def test_start_game_map_selection_newbies(
+    ladder_service: LadderService, player_factory
+):
+    p1 = player_factory(
+        ladder_rating=(1500, 500),
+        ladder_games=0,
+    )
+    p2 = player_factory(
+        ladder_rating=(300, 100),
+        ladder_games=1000,
     )
 
-    ladder_service.game_service.ladder_maps = [
-        (1, "some_map", "maps/some_map.v001.zip"),
-        (2, "some_map", "maps/some_map.v001.zip"),
-        (3, "some_map", "maps/some_map.v001.zip"),
-    ]
+    queue = ladder_service.queues["ladder1v1"]
+    queue.map_pools.clear()
+    newbie_map_pool = mock.Mock()
+    full_map_pool = mock.Mock()
+    queue.add_map_pool(newbie_map_pool, None, 500)
+    queue.add_map_pool(full_map_pool, 500, None)
 
-    chosen_map = await ladder_service.choose_map([None])
+    await ladder_service.start_game(p1, p2)
 
-    assert chosen_map is not None
+    newbie_map_pool.choose_map.assert_called_once()
+    full_map_pool.choose_map.assert_not_called()
```

Comment:
`test_start_game_map_selection` and `test_start_game_map_selection_newbies` could be one test. Use `@pytest.mark.parametrize` to configure `player_factory`s...

Reply:
OR even more, if you extract body:
```python
p1 = player_factory(
        ladder_rating=(2000, 50),
        ladder_games=1000,
    )
    p2 = player_factory(
        ladder_rating=(1500, 100),
        ladder_games=1000,
    )

    queue = ladder_service.queues["ladder1v1"]
    queue.map_pools.clear()
    newbie_map_pool = mock.Mock()
    full_map_pool = mock.Mock()
    queue.add_map_pool(newbie_map_pool, None, 500)
    queue.add_map_pool(full_map_pool, 500, None)

    await ladder_service.start_game(p1, p2)
```
You can make your test on 7 rows + 3 parametrize, 1 arguments, 1 call common method, 2 assert calls

But don't miss with assert calls on `map_pools`

PS: this comment touches also `test_start_game_map_selection_pros`